include/zephyr/bluetooth/gatt.h

```diff
@@ -167,16 +167,57 @@ typedef ssize_t (*bt_gatt_attr_write_func_t)(struct bt_conn *conn,
 
 /** @brief GATT Attribute structure. */
 struct bt_gatt_attr {
-	/** Attribute UUID */
+	/** @brief Attribute Type, aka. "UUID" */
 	const struct bt_uuid *uuid;
-	/** Attribute read callback */
+
+	/** @brief Attribute Value function
+	 *
+	 * This is the general API to read from the Attribute Value of
+	 * this Attribute in on-air format.
+	 *
+	 * This function may safely assume the Attribute Permissions are
+	 * satisfied for this read.
+	 *
+	 * This function is primarily invoked by the GATT server when a
+	 * remote GATT client performs a read, but it can also be
+	 * invoked for a local read.
+	 *
+	 * May be NULL if the attribute is not readable.
+	 */
 	bt_gatt_attr_read_func_t read;
-	/** Attribute write callback */
+
+	/** @brief Attribute Value write method
+	 *
+	 * This is the general API to write to the Attribute Value of
+	 * this Attribute in on-air format.
+	 *
+	 * This function may safely assume the Attribute Permissions are
+	 * satisfied for this write.
+	 *
+	 * May be NULL if the attribute is not writable.
+	 */
 	bt_gatt_attr_write_func_t write;
-	/** Attribute user data */
+
+	/** @brief Userdata for the read() and write()
+	 *
+	 * The meaning of this field is not specified here.
+	 *
+	 * @warning Do not assume the meaning of this field based just
+	 * on the Attribute Type. Two Attributes may share Attribute
+	 * Type but have different internal implementations.
```

Comment:
Suggest to make it less scary:
```suggestion
	 * @note Two Attributes may share Attribute Type but 
	 *       have different internal implementations.
```